Server.Logic/Managers/RecipeManager.cs

```diff
@@ -170,27 +167,125 @@ public async Task<IManagerActionResult> UnlikeRecipeAsync(int recipeId, string u
         return new ManagerActionResult(ResultEnum.OK);
     }
 
-    private async void AddAndSaveImages(Recipe recipe, RecipeViewModel recipeViewModel)
+    public async Task<IManagerActionResult<Recipe[]>> GetRecipesPagedAsync(int page, int pageSize, string searchName, int[] categoryIds)
     {
-        foreach (var image in recipeViewModel.Images)
+        IQueryable<Recipe> data = _dbInfo.Recipes;
+        if(!searchName.IsNullOrEmpty())
+            data = data.Where(x => x.Name.Contains(searchName));
+        if(categoryIds is not null && categoryIds.Length > 0) 
         {
-            var imagePath = $"./recipes/{recipe.Author}/{recipe.Name}_{recipe.Id}_{image.Name}.png";
-            //TODO File.WriteAllBytes(imagePath, image.Data);
+            data = data.Where(x => x.Categories.Any(x => categoryIds.Contains(x.Id)));
+        }
+        return new ManagerActionResult<Recipe[]>(await _pageingManager.GetPagedInfo(data, page, pageSize).ToArrayAsync(),ResultEnum.OK);
+    }
 
-            var newImage = new RecipiesImage
+    public async Task<IManagerActionResult> AddReferenceToRecipeAsync(int id, int referenceId, double multiplayer)
+    {
+        await _dbInfo.RecipeChildren.AddAsync(
```

Comment:
Check na użytkownika, tylko właściciel powinien móc to dodawać, więc musisz tu przekazać usera z Controllera

Reply:
Poprawiono we wszystkich miejscach